packages/notte-agent/src/notte_agent/falco/agent.py

```diff
@@ -297,8 +309,12 @@ async def _run(self, task: str, url: str | None = None) -> AgentResponse:
         if url is not None:
             task = f"Start on '{url}' and {task}"
 
-        # Loop through the steps
-        async with self.session:
+        started_before_run = self.session.started
```

Comment:
I don't like this, is this really needed ? I would prefer to raise an exception is the session is not started, especially if the session comes from the sdk. Otherwise the users don't know the sessions have been started and they have no way to know they should close it